src/main/scala/tilelink/Fragmenter.scala

```diff
@@ -57,7 +57,8 @@ class TLFragmenter(val minSize: Int, val maxSize: Int, val alwaysMin: Boolean =
     clientFn  = { c => c.copy(clients = Seq(TLClientParameters(
       name        = "TLFragmenter",
       sourceId    = IdRange(0, if (minSize == maxSize) c.endSourceId else (c.endSourceId << addedBits)),
-      requestFifo = true))) },
+      requestFifo = true,
+      userBits    = c.clients.map( _.userBits).maxBy(_.length))))},
```

Comment:
Hmm, making the decision based on `length` doesn't seem necessarily correct, because you are looking at the number of different types of user bits on the link, but not the size of each type. Poking around I found:
https://github.com/chipsalliance/rocket-chip/blob/master/src/main/scala/tilelink/Parameters.scala#L260
... which I think would be convenient for use with`maxBy`, but overall this raises questions for me about correct behavior when the clients dont match.

@terpstra Should mismatches between `TLClient`s' `userBits` here fail a requirement?

Reply:
`userBits` is of type `Seq[UserBits]`. So the map returns a `Seq[Seq[UserBits]]`. Shouldn't we be selecting the `Seq[UserBits]` of the maximum length from this mapped Seq?

Reply:
As we discussed, the maximum bit length and maximum sequence length are not synonymous.

I talked with @terpstra offline and we agreed that for now the safe thing to do is have a requirement checking that all clients have identical userBits Seqs, and to fail if they do not match. (This strictness will require us to do more work for the target design to add user bit generation to more masters, but so be it.)

Reply:
By identical, I'm guessing you mean the bit-widths should also match and in order?

Reply:
Yes.

Also, I just spoke with Kenneth. I think that the proposal @hcook outlines will play nicely with another block we need anyway: a UserBits extractor adaptor. The idea here is that right before the TLToAPB/AHB you put down a TLUserExtract[T](x) that selects exactly one kind of user bits field and supplies a default value for any masters that had none. This has to be done on the TileLink side of the adapter, because AHB/APB have no way to identify the source of a request.